api/src/org/apache/cloudstack/api/command/admin/template/PrepareTemplateCmd.java

```diff
@@ -60,6 +61,15 @@
                description = "template ID of the template to be prepared in primary storage(s).")
     private Long templateId;
 
+    @ACL(accessType = AccessType.OperateEntry)
+    @Parameter(name = ApiConstants.STORAGE_ID,
+            type = CommandType.UUID,
+            entityType = StoragePoolResponse.class,
+            required = false,
+            description = "storage pool ID of the primary storage pool to which the template should be prepared. If it is not provided the template" +
+                    " is prepared on all the available primary storage pools.")
+    private Long storageId;
```

Comment:
@remibergsma, @kishankavala is concurring with the observation by @devdeep that his call was good in the name of consistency. I think he was being unconditional about his LGTM.